src/serverenvironment.cpp

```diff
@@ -1855,22 +1833,14 @@ void ServerEnvironment::activateObjects(MapBlock *block, u32 dtime_s)
 		// This will also add the object to the active static list
 		addActiveObjectRaw(obj, false, dtime_s);
 	}
+
 	// Clear stored list
 	block->m_static_objects.m_stored.clear();
 	// Add leftover failed stuff to stored list
 	for (const StaticObject &s_obj : new_stored) {
 		block->m_static_objects.m_stored.push_back(s_obj);
 	}
 
-	// Turn the active counterparts of activated objects not pending for
-	// deactivation
-	for (auto &i : block->m_static_objects.m_active) {
-		u16 id = i.first;
-		ServerActiveObject *object = getActiveObject(id);
-		assert(object);
-		object->m_pending_deactivation = false;
```

Comment:
since the SAOs are newly created in the `for` loop above, they assume the default value...
...which is `false`
-> no-op